### PR TITLE
chores: fixed minor issues, prepare for merging into main

### PR DESCRIPTION
**Pull request by ivorkchan**
## Related Issue

Fixes # (issue)

## Type of Change
- [ ] Bug fix
- [ ] New feature
- [x] Refactoring
- [x] Documentation update
- [ ] Test update
- [x] CI/CD pipeline update
- [x] Other changes

## Checklist
- [x] I have reviewed my code to ensure it meets project standards.
- [x] I have added comments in complex or non-obvious parts of the code.
- [x] I have updated `README.md` (for new features) if applicable.
- [x] I have updated the project changelog (for each change).
- [x] I have included tests or test updates that verify my changes.
- [x] I have ensured that new and existing unit tests pass locally.
- [x] I have addressed `CheckStyle` warnings in my code.
- [x] I have requested a review from another team member.

## Reviewer Checklist
- [x] I have verified that all unit tests pass successfully.
- [x] I have ensured that documentation, including README and changelog, is updated.
- [x] I have checked that the code follows dependency rules specified by Clean Architecture.
- [x] I have confirmed that comments are clear, concise, and current.
- [x] I have verified the code is free of hardcoded values, unnecessary complexity, or unused code.

### CODE REVIEW

File: src/main/java/use_case/execute_buy/ExecuteBuyInteractor.java [hunk 1]
```diff
@@ -74,19 +73,18 @@ private void updateOrAddStockToPortfolio(Portfolio portfolio, Stock stock, int q
 
     private void handleException(String message) {
```
Comment (Raine-Yang-UofT) [inline src/main/java/use_case/execute_buy/ExecuteBuyInteractor.java:74]: I'm not sure separating a helper method really simplifies the logic, as we still need to pass in a unique string for every error and make sure the string messages on both sides match. I suggest we can simply invoke the `ouputPresenter` in the `catch` statements.

File: src/main/java/use_case/execute_buy/ExecuteBuyDataAccessInterface.java [hunk 2]
```diff
@@ -3,9 +3,9 @@
 import entity.User;
 
 public interface ExecuteBuyDataAccessInterface {
-    User getUserWithCredential(String credential) throws ValidationException;
+    User getUserWithCredential(String credential) throws ValidationError;
 
     // TODO: define this exception elsewhere
-    class ValidationException extends Exception {
+    class ValidationError extends Exception {
```
Comment (Raine-Yang-UofT) [inline src/main/java/use_case/execute_buy/ExecuteBuyDataAccessInterface.java:9]: Change this exception name back to `ValidationException`. In `Java`, `Error` usually implies a subclass of `Error`, a type used to indicate a serious system-level error that cannot be recovered (such as out of memory). `Exception` indicates subclass of `Exception`, an exception intended to be handled in the program.

File: src/main/java/use_case/execute_buy/ExecuteBuyInteractor.java (hunk 1 above)
Comment (ivorkchan) [reply src/main/java/use_case/execute_buy/ExecuteBuyInteractor.java:74]: I'm not sure whether the session manager implementation will change in the future, but now it's OK to let each presenter handle that. I'll delete the todo.

File: src/main/java/interface_adapter/execute_buy/ExecuteBuyPresenter.java [hunk 3]
```diff
@@ -14,23 +14,18 @@ public void prepareSuccessView(ExecuteBuyOutputData outputData) {
...
     }
 
     @Override
-    public void prepareStockNotFoundView() {
-
+    public void prepareStockNotFoundErrorView() {
     }
 
     @Override
     public void prepareValidationErrorView() {
```
Comment (Raine-Yang-UofT) [inline src/main/java/interface_adapter/execute_buy/ExecuteBuyPresenter.java:29]: Refer to my comment below. `Error` and `Exception` are two different concepts and should not be misused.

**Review by Raine-Yang-UofT — APPROVED** [review]

**Pull request merged**